### PR TITLE
Enable transfer env variables to engine launcher process

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_
Enable transfer some env variables to engine launcher process

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
A list may be hard to use and cause some parsing issues for  the JDBC URL, can we make it simpler? Users can use `kyuubi.engineEnv.VAR_NAME`, then provides a `getEnvs` in `KyuubiConf` to retrieve all environment vars

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -101,6 +101,13 @@ case class KyuubiConf(loadSysDefault: Boolean = true) extends Logging {
     settings.entrySet().asScala.map(x => (x.getKey, x.getValue)).toMap[String, String]
   }
 
+  /** Get all engine envs as map */
+  def getEngineEnvs: Map[String, String] = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:105]: use getAllWithPrefix?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -101,6 +101,13 @@ case class KyuubiConf(loadSysDefault: Boolean = true) extends Logging {
...
+    settings.entrySet().asScala.filter(_.getKey.startsWith(KYUUBI_ENGINE_ENV_PREFIX))
+      .map(entry => entry.getKey.stripPrefix(KYUUBI_ENGINE_ENV_PREFIX) -> entry.getValue)
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:107]: use getAllWithPrefix?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:105]: `getEnvs` can be more general name

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:107]: this should also lookup `sys.env` first and apply `KYUUBI_ENGINE_ENV_PREFIX`s in the rightmost way

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.3.0
